railties/lib/rails/generators/rails/resource_route/resource_route_generator.rb

```diff
@@ -21,7 +25,9 @@ def add_resource_route
         end
 
         # inserts the primary resource
-        write("resources :#{file_name.pluralize}", route_length + 1)
+        resource_string = "resources :#{file_name.pluralize}"
+        resource_string << ", except: [:new, :edit]" if options.api?
```

Comment:
@vipulnsward @rafaelfranca Hey guys, let me share a few thoughts

- Regarding the routes generated by `resources` in API only mode, the reason for this behavior is to avoid having a `routes.rb` file where all the `resources` have the `except: [:new, :edit]`. Please considerar that API only apps should be applications where all the controllers inherits from AC::API and having actions like `new` and `edit` never makes sense. For this reason, it sounded odd to have to put `except: [:new, :edit]` in many places to achieve something that could be deduced automatically.

- This PR is related to the usage of AC::API in a regular Rails app (where api_only flag is not set). We definitively can support this special usage, but I think it is not a great idea to have a way to generate AC::API controllers via generators in a regular app. I'd remove the --api at all from scaffold generators and change the generator code to rely on the app config `api_only` flag directly.

- Related to the previous point, a `--no-api` option currently exists in the generators ([it's created automatically by thor](https://github.com/erikhuda/thor/blob/master/lib/thor/parser/option.rb#L89)). So, in an only API app AC::Base controllers can be generated, but these controllers won't work at all because an only API app does not have all the necessary middlewares to work with this kind of controllers. It's an issue that should be fixed, and it also can be fixed by removing the `--api` option.

- Regarding only API apps and engines: I haven't had much time to think in all the cases, but I can affirm that mounting a normal engine in an only API app is not going to work because of the middlewares. The host app and all the engines share the same middleware stack, and this stack is defined by the api_only flag in the application, so an engine including regular controllers and using stuff like ActionView is going to fail. I don't know how we should prevent the usage of regular engines in only API apps but it has to be done somehow (at least while Rails apps have a unique shared middleware stack for the application itself and engines 😸).